packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/DevLauncherController.kt

```diff
@@ -148,7 +148,7 @@ class DevLauncherController private constructor() :
       val manifestParser = DevLauncherManifestParser(httpClient, parsedUrl, installationIDHelper.getOrCreateInstallationID(context))
       val appIntent = createAppIntent()
 
-      internalUpdatesInterface?.reset()
+      DevLauncherKoinContext.app.koin.getOrNull<UpdatesInterface>()?.reset()
```

Comment:
@lukmccall since we changed the initialization order, the `DevLauncherController.instance.updatesInterface = it` will be called after this function. the default koin injection for `internalUpdatesInterface` would make it as null even later `DevLauncherController.instance.updatesInterface = it` is called.

that's the main reason to change it as `DevLauncherKoinContext.app.koin.getOrNull<UpdatesInterface>()?.reset()` here.
not sure whether the change makes sense to you?

the other question, do you know whether we still use `internalUpdatesInterface` in our code? it looks like we don't use it in [any other places.](https://github.com/search?q=repo%3Aexpo%2Fexpo%20internalUpdatesInterface&type=code) maybe we can also remove `internalUpdatesInterface` without using DI?

Reply:
yep let's keep it as-is without removing `internalUpdatesInterface`. we can do it when it's necessary for a refactoring.